File: src/main/java/catchla/yep/fragment/TopicChatListFragment.java
package catchla.yep.fragment;

import android.accounts.Account;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.content.Loader;

import java.util.List;

import catchla.yep.loader.HistoricalMessagesLoader;
import catchla.yep.model.Message;
import catchla.yep.model.Paging;
import catchla.yep.model.Topic;
import catchla.yep.util.YepAPI;

/**
 * Created by mariotaku on 15/12/10.
 */
public class TopicChatListFragment extends ChatListFragment {
    @Override
    public void onLoadFinished(final Loader<List<Message>> loader, final List<Message> data) {
        super.onLoadFinished(loader, data);
        setRefreshEnabled(false);
    }

    @Override
    public void onActivityCreated(@Nullable final Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        setRefreshEnabled(false);
    }

    @Override
    public Loader<List<Message>> onCreateLoader(final int id, final Bundle args) {
        final Topic topic = getTopic();
        final String recipientId = topic.getCircle().getId();
        return new HistoricalMessagesLoader(getContext(), getAccount(),
                YepAPI.PathRecipientType.CIRCLES, recipientId, new Paging(), false, false,
                getAdapter().getData());
    }

    public Topic getTopic() {
        return getArguments().getParcelable(EXTRA_TOPIC);
    }

    public Account getAccount() {
        return getArguments().getParcelable(EXTRA_ACCOUNT);
    }
}

File: src/main/kotlin/catchla/yep/fragment/TopicChatListFragment.kt
package catchla.yep.fragment

import android.os.Bundle
import android.support.v4.content.Loader
import catchla.yep.annotation.PathRecipientType
import catchla.yep.extension.account
import catchla.yep.loader.HistoricalMessagesLoader
import catchla.yep.model.Message
import catchla.yep.model.Paging

/**
 * Created by mariotaku on 15/12/10.
 */
class TopicChatListFragment : ChatListFragment() {
    override fun onLoadFinished(loader: Loader<List<Message>?>, data: List<Message>?) {
        super.onLoadFinished(loader, data)
        refreshEnabled = false
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        refreshEnabled = false
    }

    override fun onCreateLoader(id: Int, args: Bundle?): Loader<List<Message>?> {
        val topic = topic!!
        val recipientId = topic.circle.id
        return HistoricalMessagesLoader(context, account, PathRecipientType.CIRCLES,
                recipientId, Paging(), false, false, adapter.data)
    }

}
